evalml/pipelines/components/transformers/scalers/standard_scaler.py

```diff
@@ -26,10 +25,9 @@ def __init__(self, random_seed=0, **kwargs):
 
     def transform(self, X, y=None):
         X_ww = infer_feature_types(X)
-        X = _convert_woodwork_types_wrapper(X_ww.to_dataframe())
         X_t = self._component_obj.transform(X)
-        X_t_df = pd.DataFrame(X_t, columns=X.columns, index=X.index)
-        return _retain_custom_types_and_initalize_woodwork(X_ww, X_t_df, ltypes_to_ignore=[Integer, Categorical])
+        X_t_df = pd.DataFrame(X_t, columns=X_ww.columns, index=X_ww.index)
+        return _retain_custom_types_and_initalize_woodwork(X_ww.ww.logical_types, X_t_df, ltypes_to_ignore=[Integer, Categorical, Boolean])
```

Comment:
If you pass in a `Boolean` to `StandardScaler` you get back a float. In the old version of woodwork, you can't convert from float to boolean because of the nullable pandas types. In this version of woodwork, that conversion will succeed. 

In order to preserve the current behavior, we need to ignore booleans now.